MREUnityRuntime/MREUnityRuntimeLib/App/MixedRealityExtensionApp.cs

```diff
@@ -577,8 +577,12 @@ private void OnCreateFromPrefab(CreateFromPrefab payload, Action onCompleteCallb
         {
             try
             {
-                var createdActors = _assetLoader.CreateFromPrefab(payload.PrefabId, payload.Actor?.ParentId);
-                ProcessCreatedActors(payload, createdActors, onCompleteCallback);
+                MREAPI.AppsAPI.AssetCache.OnCached(payload.PrefabId, _ =>
+                {
+                    if (!_ownerScript) return;
```

Comment:
_ownerScript doesn't get destroyed when the app disconnects. I think you'll need to do this another way. One idea: Keep a counter and increment it each time Conn_Disconnected is called. Check that the counter has the expected value in the callback.

Reply:
How about `this == null`?

---
In reply to: [307060178](https://github.com/microsoft/mixed-reality-extension-unity/pull/92#discussion_r307060178) [](ancestors = 307060178)

Reply:
That doesn't work either. This object doesn't get destroyed on a disconnect.

---
In reply to: [307061161](https://github.com/microsoft/mixed-reality-extension-unity/pull/92#discussion_r307061161) [](ancestors = 307061161,307060178)

Reply:
I would be surprised if checking _appState == AppState.Running will work. Did you test it with a second peer joining and syncing assets? AppState isn't set to running until after synchronization completes, and assets load before that time.

---
In reply to: [307063204](https://github.com/microsoft/mixed-reality-extension-unity/pull/92#discussion_r307063204) [](ancestors = 307063204,307061161,307060178)

Reply:
Also, the app could have gone from running -> stopped -> running while the prefab was loading. You need to know for certain in the callback that the websocket didn't disconnect and reconnect.

---
In reply to: [307066354](https://github.com/microsoft/mixed-reality-extension-unity/pull/92#discussion_r307066354) [](ancestors = 307066354,307063204,307061161,307060178)